egs/iam/v2/run_end2end.sh

```diff
@@ -27,56 +29,97 @@ wellington_database=/export/corpora5/Wellington/WWC/
 ./local/check_tools.sh
 
 if [ $stage -le 0 ]; then
+  if [ -f data/train/text ] && ! $overwrite; then
+    echo "Not processing, probably script have run from wrong stage"
```

Comment:
Don't forget `$0`. Also I'm not sure how this will help if someone calls this script from stage 1, maybe you meant to put this check in stage 1 (before `echo "$0: Preparing data..."`)?